src/late_multicellular_stage/MulticellularConvolutionDispayer.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   Creature component to display it using convolution surfaces
+/// </summary>
+public partial class MulticellularConvolutionDispayer : MeshInstance3D, IMetaballDisplayer<MulticellularMetaball>
+{
+    private const float AABBMargin = 0.1f;
+
+    private StandardMaterial3D? material;
+
+    private float? overrideColourAlpha;
+
+    public float? OverrideColourAlpha
+    {
+        get => overrideColourAlpha;
+        set
+        {
+            // Due to both being nullable, this would be a bit complicated to compare with an epsilon value
+            // ReSharper disable once CompareOfFloatsByEqualityOperator
+            if (overrideColourAlpha == value)
+                return;
+
+            overrideColourAlpha = value;
+            ApplyAlpha();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        // This is here in case we need custom shader effects at some point
+        // material = new ShaderMaterial()
+        // {
+        //     Shader = GD.Load<Shader>("res://shaders/Metaball.shader"),
+        // },
+        material = new StandardMaterial3D
+        {
+            VertexColorUseAsAlbedo = true,
+        };
+
+        ApplyAlpha();
+
+        ExtraCullMargin = AABBMargin;
+    }
+
+    public void DisplayFromList(IReadOnlyCollection<MulticellularMetaball> layout)
+    {
+        Vector3 minExtends = Vector3.Zero;
+        Vector3 maxExtends = Vector3.Zero;
+
+        foreach (var metaball in layout)
+        {
+            minExtends.X = Mathf.Min(minExtends.X, metaball.Position.X - metaball.Radius - 0.5f);
+            minExtends.Y = Mathf.Min(minExtends.Y, metaball.Position.Y - metaball.Radius - 0.5f);
+            minExtends.Z = Mathf.Min(minExtends.Z, metaball.Position.Z - metaball.Radius - 0.5f);
+
+            maxExtends.X = Mathf.Max(maxExtends.X, metaball.Position.X + metaball.Radius + 0.5f);
+            maxExtends.Y = Mathf.Max(maxExtends.Y, metaball.Position.Y + metaball.Radius + 0.5f);
+            maxExtends.Z = Mathf.Max(maxExtends.Z, metaball.Position.Z + metaball.Radius + 0.5f);
+        }
+
+        var meshGen = new DualContourer();
```

Comment:
Could you add a TODO comment here saying that these expensive calculations really need a cache in the future.

For now this is fine as long as it doesn't freeze the game for too long, but in the long run we'll need a proper generation time for when to do the expensive calculations to not freeze during gameplay when a new creature spawns.